extensions/kusto/src/credentialstore/credentialstore.ts

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SqlOpsDataClient, ClientOptions, SqlOpsFeature } from 'dataprotocol-client';
+import { IConfig, ServerProvider } from 'service-downloader';
+import { ServerOptions, RPCMessageType, ClientCapabilities, ServerCapabilities, TransportKind } from 'vscode-languageclient';
+import { Disposable } from 'vscode';
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+import * as azdata from 'azdata';
+
+import * as Contracts from './contracts';
+import * as Constants from './constants';
+import * as Utils from '../utils';
+
+class CredentialsFeature extends SqlOpsFeature<any> {
+
+	private static readonly messagesTypes: RPCMessageType[] = [
+		Contracts.DeleteCredentialRequest.type,
+		Contracts.SaveCredentialRequest.type,
+		Contracts.ReadCredentialRequest.type
+	];
+
+	constructor(client: SqlOpsDataClient) {
+		super(client, CredentialsFeature.messagesTypes);
+	}
+
+	fillClientCapabilities(capabilities: ClientCapabilities): void {
+		Utils.ensure(Utils.ensure(capabilities, 'credentials')!, 'credentials')!.dynamicRegistration = true;
+	}
+
+	initialize(capabilities: ServerCapabilities): void {
+		this.register(this.messages, {
+			id: UUID.generateUuid(),
+			registerOptions: undefined
+		});
+	}
+
+	protected registerProvider(options: any): Disposable {
+		const client = this._client;
+
+		let readCredential = (credentialId: string): Thenable<azdata.Credential> => {
+			return client.sendRequest(Contracts.ReadCredentialRequest.type, { credentialId, password: undefined });
+		};
+
+		let saveCredential = (credentialId: string, password: string): Thenable<boolean> => {
+			return client.sendRequest(Contracts.SaveCredentialRequest.type, { credentialId, password });
+		};
+
+		let deleteCredential = (credentialId: string): Thenable<boolean> => {
+			return client.sendRequest(Contracts.DeleteCredentialRequest.type, { credentialId, password: undefined });
+		};
+
+		return azdata.credentials.registerProvider({
+			deleteCredential,
+			readCredential,
+			saveCredential,
+			handle: 0
+		});
+	}
+}
+
+/**
+ * Implements a credential storage for Windows, Mac (darwin), or Linux.
+ *
+ * Allows a single credential to be stored per service (that is, one username per service);
+ */
+export class CredentialStore {
+	private _client: SqlOpsDataClient;
+	private _config: IConfig;
+
+	constructor(private logPath: string, baseConfig: IConfig) {
+		if (baseConfig) {
+			this._config = JSON.parse(JSON.stringify(baseConfig));
+			this._config.executableFiles = ['MicrosoftSqlToolsCredentials.exe', 'MicrosoftSqlToolsCredentials'];
```

Comment:
We'd be running two instances of this per ADS run. It's going to make debugging this an extreme pain since we'd have to guess which one is the mssql one.

Recommend what Karl said and extract this out.

(blocking issue)

Reply:
Removed the usage of this file as well.